privacyidea/lib/tokens/radiustoken.py

```diff
@@ -137,11 +144,174 @@ def update(self, param):
         TokenClass.update(self, param)
         val = getParam(param, "radius.local_checkpin", optional) or 0
         self.add_tokeninfo("radius.local_checkpin", val)
-
         val = getParam(param, "radius.user", required)
         self.add_tokeninfo("radius.user", val)
         self.add_tokeninfo("tokenkind", TOKENKIND.VIRTUAL)
 
+    @log_with(log)
+    def is_challenge_request(self, passw, user=None, options=None):
+        """
+        This method checks, if this is a request, that triggers a challenge.
+        It depends on the way, the pin is checked - either locally or remote
+
+        :param passw: password, which might be pin or pin+otp
+        :type passw: string
+        :param user: The user from the authentication request
+        :type user: User object
+        :param options: dictionary of additional request parameters
+        :type options: dict
+
+        :return: true or false
+        """
+        if options is None:
+            options = {}
+
+        # should we check the pin locally?
+        if self.check_pin_local:
+            # With a local PIN the challenge response is always a privacyIDEA challenge response!
+            res = self.check_pin(passw)
+
+        else:
+            state = options.get('radius_state')
+            # The pin is checked remotely
+            res = options.get('radius_result')
+            if res is None:
+                res = self._check_radius(passw, options=options, radius_state=state)
+
+            return res == AccessChallenge
+
+        return res
```

Comment:
Could we move this to the ``if`` branch body?

Reply:
done.